### PR TITLE
[autoconf/automake/libtool] Add new port

### PR DESCRIPTION
**Pull request by JackBoosY**
The tools `autoconf` and `automake` are used in `vcpkg_*_make`, and I want to add these two ports instead of using apt to install them.

Thanks @SeekingMeaning.

Related: #9648.

Note: these 3 ports does not contain any features.

### CODE REVIEW

**Comment by SeekingMeaning** [comment]
Nice. I would recommend adding libtool as well as it is part of the autotools suite. See https://www.gnu.org/software/automake/faq/autotools-faq.html#What-are-the-_0060_0060Autotools_0027_0027_003f

File: ports/libtool/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+vcpkg_fail_port_install(ON_TARGET "Linux")
```
Comment (SeekingMeaning) [inline ports/libtool/portfile.cmake]: `Linux` hmmm

**Comment by JackBoosY** [comment]
After merging this PR, I will open a new PR to switch usage to these ports.

**Review by PhoebeHui — APPROVED** [review]

**Comment by vicroms** [comment]
@JackBoosY 

These should really be handled by `vcpkg_find_acquire_program`, if we make these tools into ports, they will be built for the target platform. Currently, we lack a mechanism to have "tool ports" that are built for the host platform.

**Comment by JackBoosY** [comment]
@vicroms In addition to the way of using, adding these tools by using ports will reduce the dependency on `apt` / `brew` / `yum`. We already have this example (e.g. llvm),

**Comment by Neumann-A** [comment]
vcpkg should really have the field: `Build-Tools: ` which is similar to `Build-Depends: ` with the important difference that `Build-Tools: ` is using `VCPKG_HOST_TRIPLET`. If `VCPKG_HOST_TRIPLET` is not defined it is the same as `VCPKG_TARGET_TRIPLET`.

**Comment by dan-shaw** [comment]
@JackBoosY Would it be okay to archive this PR for now? We do want to add tools as ports but not likely in the near future.

**Comment by JackBoosY** [comment]
@dan-shaw Okay, I will not delete this branch until these ports are merged into the main branch.

**Comment by dg0yt** [comment]
With host dependencies in vcpkg, does it make sense to revive this PR? 
There is a lot of bad UX around bootstrapping with brew, apt or whatever.

**Comment by JackBoosY** [comment]
@dg0yt  Sooner or later we have to break away from the limitations of apt and brew.

**Pull request closed without merging**